### DISCUSSION TITLE
Debugging Argo Notifications

### DISCUSSION THREAD

**Discussion by lukonjun** [argoproj/argo-cd#9232] [Q&A]
Hi there, i want to use Argo Notification in combination with Slack, I created the following configuration trying to trigger just anything however I cant see any change in the logs
```
time="2022-04-27T20:25:56Z" level=info msg="Start processing" resource=argocd/frontend-webapp-pr-27
time="2022-04-27T20:25:56Z" level=info msg="Trigger app-sync-status result: []" resource=argocd/frontend-webapp-pr-27
time="2022-04-27T20:25:56Z" level=info msg="Processing completed" resource=argocd/frontend-webapp-pr-27
time="2022-04-27T20:25:56Z" level=info msg="Start processing" resource=argocd/frontend-webapp-pr-3
time="2022-04-27T20:25:56Z" level=info msg="Trigger app-sync-status result: []" resource=argocd/frontend-webapp-pr-3
time="2022-04-27T20:25:56Z" level=info msg="Processing completed" resource=argocd/frontend-webapp-pr-3
```
I am using this configuration and trying to deploy, sync and destroy applications to trigger any notifications
```
apiVersion: v1
kind: ConfigMap
metadata:
  name: argocd-notifications-cm
data:
  trigger.app-sync-status: |
    - when: app.status.operationState.phase in ['Succeeded']
      send: [app-sync-status]
    - when: app.status.sync.status == 'Unknown'
      send: [app-sync-status] 
    - when: app.status.operationState.phase in ['Succeeded']
      send: [app-sync-status] 
    - when: app.status.sync.status == 'Unknown'
      send: [app-sync-status] 
    - when: app.status.operationState.phase in ['Running']
      send: [app-sync-status] 
    - when: app.status.operationState.phase in ['Error', 'Failed']
      send: [app-sync-status] 
    - when: app.status.health.status == 'Degraded'
      send: [app-sync-status] 

  service.slack: |
    token: $slack-token

  template.app-sync-status: |
    message: |
      Application {{.app.metadata.name}} sync is {{.app.status.sync.status}}.
      Application details: {{.context.argocdUrl}}/applications/{{.app.metadata.name}}.
    slack:
      attachments: |
        [{
          "title": "{{.app.metadata.name}}",
          "title_link": "{{.context.argocdUrl}}/applications/{{.app.metadata.name}}",
          "color": "#18be52",
          "fields": [{
            "title": "Sync Status",
            "value": "{{.app.status.sync.status}}",
            "short": true
          }, {
            "title": "Repository",
            "value": "{{.app.spec.source.repoURL}}",
            "short": true
          }]
        }]

```
Am I miss reading the logs or not correctly triggering a change, thanks in advance!

**Comment by nanoq66** [comment]
yes the log should contain something like this (not the filled [] with the result)

│ time=“2022-11-21T16:03:04Z” level=info msg=“Trigger on-sync-failed result: [{[0].WhDttPykq_Az2vhfHWLgbRUZ2d0 [app-sync-failed] false}]”

**Comment by abhipsnl** [comment]
Same issue, ant luck on this?

**Comment by lukonjun** [comment]
Hi @sharmajee1, I checked out this feature a couple of month ago, not using it anymore, but remember I made it work. This this discussion might help with the error https://github.com/argoproj/argo-cd/discussions/9243

The answer was found in this slack discussion. Hope this help https://cloud-native.slack.com/archives/C01UKS2NKK3/p1651230087597369
> The configmap and secret I already had in the namespace were called argocd-notifications-secret and argocd-notifications-cm  so it meant they weren't being referenced by my new notifications deployment. Instead the new deployment was referencing a secret and cm with -controller in the name.
> This new secret and cm didn't have my slack token or templates in so I had to revisit my values.yaml and specify the name of the secret to be argocd-notifications-secret and added in the templates too
